source/includes/fundamentals/code-snippets/Monitoring.java

```diff
@@ -0,0 +1,151 @@
+                        .addCommandListener(new CommandCounter())
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        // Run some commands to test the timer
+        collection.find().first();
+        collection.find().first();
+        mongoClient.close();
+        // end monitor-command-example
+    }
+
+    private void monitorClusterEvent() {
+        // start monitor-cluster-example
+        IsWriteable clusterListener = new IsWriteable();
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .applyToClusterSettings(builder ->
+                                builder.addClusterListener(clusterListener))
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        // Run a command to trigger a ClusterDescriptionChangedEvent event
+        collection.find().first();
+        // end monitor-cluster-example
+        mongoClient.close();
+    }
+
+    private void monitorConnectionPoolEvent() {
+        // start monitor-cp-example
+        ConnectionPoolLibrarian cpListener = new ConnectionPoolLibrarian();
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .applyToConnectionPoolSettings(builder ->
+                                builder.addConnectionPoolListener(cpListener))
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        // Run a command to trigger connection pool events
+        collection.find().first();
+        // end monitor-cp-example
+        /* We do not close this connection in order to prevent the driver from requesting two connections, giving
+        the example unintuitive output. Uncomment the following line of code and see how it effects this example*/
+        //mongoClient.close();
+    }
+
+}
+
+// start command-listener-impl
+class CommandCounter implements CommandListener {
+
+    private Map<String, Integer> commands = new HashMap<String, Integer>();
+
+    @Override
+    public void commandSucceeded(final CommandSucceededEvent event) {
+        String commandName = event.getCommandName();
+        int count = commands.containsKey(commandName) ? commands.get(commandName) : 0;
+        commands.put(commandName, count + 1);
+        System.out.println(commands.toString());
```

Comment:
Use `commands.toJson()` instead (this will change the output to JSON, so you'll have to change that as well.

Reply:
Sorry, I mis-read this as `System.out.println(command.toString()`

Reply:
Ok, sounds good, I'll revert this back to the style it was when the initial comment in this thread was posted:

 `System.out.println(commands.toString());`

Reply:
So you should revert this change.

Reply:
Reverting Changes: https://github.com/mongodb/docs-java/pull/88/commits/72d0bb1157dc8c7b3d16a5c287e6e132251c6a0e